.github/workflows/heads.yml

```diff
@@ -0,0 +1,51 @@
+name: heads
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Specs - Ruby ${{ matrix.ruby }} ${{ matrix.appraisal }}${{ matrix.name_extra || '' }}
+    env: # $BUNDLE_GEMFILE must be set at job level, so it is set for all steps
+      BUNDLE_GEMFILE: ${{ github.workspace }}/${{ matrix.gemfile }}.gemfile
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental || endsWith(matrix.ruby, 'head') }}
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          # ruby-head
+          - ruby: "head"
+            appraisal: "rails-8-0"
+            taskname: "spec"
+            gemfile: "appraisal_root"
+            rubygems: latest
+            bundler: latest
+
+          # truffleruby-head
+          - ruby: "truffleruby-head"
+            appraisal: "rails-8-0"
+            taskname: "spec"
+            gemfile: "appraisal_root"
+            rubygems: latest
+            bundler: latest
```

Comment:
Building against `head` is just a canary, an early warning about upcoming technical debt when you discover things will potentially break in the next Ruby release, once it is released.

IMO it is most useful as a discrete workflow because it should be allowed to fail, and while [GitHub doesn't have a proper "allow failures" feature](https://github.com/actions/runner/issues/2347#issuecomment-2301892819), they should, and they will.

Reply:
I'll remove the truffleruby-head.  I add it to my open source gems whenever I can because I like to support it.

Reply:
By the way, this church getting the CI to work is because your setup is different from the type I usually run (I always add development dependencies to the gemspec (in violation of the default RuboCop rule!) because it simplifies things for me... I'll have it working shortly.